app/src/app/components/login/DeviceLogin.tsx

```diff
@@ -0,0 +1,83 @@
+import VerificationInput from "react-verification-input";
+import { Button } from "../Base/Button";
+import appConfig from "../../../config/appConfig";
+import { HttpStatus } from "../../../lib/types/HttpStatus";
+import { useState } from "react";
+import { getAuthHeader } from "../../../lib/auth/getAuthHeader";
+import { Check } from "lucide-react";
+
+export const DeviceLogin = () => {
```

Comment:
best to use react-hook-form and exisitng form components with https://ui.shadcn.com/docs/components/input-otp as fields... you can check other forms such as `AddRoleForm.tsx` for inspiration.

this will clean a lot of manual form handling logic up and worrying about state.. also will be consistent with rest of app